File: frontend/where_child_bus/lib/util/health_check.dart
import 'dart:async';
import 'dart:developer' as developer; // エラーログに使用
import 'package:grpc/grpc.dart';
import 'package:where_child_bus/proto-gen/where_child_bus/v1/health_check.pbgrpc.dart';


Future<PingResponse> createHoge() async {
  final channel = ClientChannel(
    'where-child-bus-grpc-k3dkun2lpq-uw.a.run.app',
    port: 443,
    options: const ChannelOptions(credentials: ChannelCredentials.insecure()),
  );
  final grpcClient = HealthcheckServiceClient(channel, options: CallOptions(timeout: const Duration(seconds: 10)));

  try {
    PingResponse res = await grpcClient.ping(PingRequest()..name = "PING");
    await channel.shutdown();
    return Future<PingResponse>.value(res);
  } catch (error) {
    developer.log('Caught error: $error');
    await channel.shutdown();
    return Future.error(error);
  }
}

File: machine_learning/src/proto-gen/where_child_bus/v1/healthcheck.py
import os
import sys

# このスクリプトが存在するディレクトリの親ディレクトリを検索パスに追加
sys.path.append(os.path.dirname(os.path.dirname(os.path.abspath(__file__))))

import grpc
import health_check_pb2
import health_check_pb2_grpc

# gRPCクライアントの設定
client = grpc.insecure_channel("where-child-bus-grpc-k3dkun2lpq-uw.a.run.app:443")
stub = health_check_pb2_grpc.HealthcheckServiceStub(client)

reqst = health_check_pb2.PingRequest()
reqst.name = "ping"

res = health_check_pb2.PingResponse(stub.Ping(reqst))

print(res.message)
